dashboard/src/components/util/SigninRedirect.js

```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import { Route, Redirect } from "react-router-dom";
+
+const signinPath = "/signin";
+const redirectKey = "redirect-to";
```

Comment:
"redirect-to" is defined in several places. I suggest defining it in a single location.

This would extend nicely to _any_ query params - since it is a single namespace, we can define all param names in a single location to avoid potential conflicts.

Reply:
Good thinking, addressed.